src/backends/r1cs/mod.rs

```diff
@@ -357,16 +337,16 @@ where
     fn finalize_circuit(
         &mut self,
         public_output: Option<crate::var::Var<Self::Field, Self::Var>>,
-        returned_cells: Option<Vec<R1csCellVar<F>>>,
-        main_span: crate::constants::Span,
+        returned_cells: Option<Vec<LinearCombination<F>>>,
+        main_span: Span,
     ) -> crate::error::Result<()> {
         // store the return value in the public input that was created for that
         if let Some(public_output) = public_output {
             let cvars = &public_output.cvars;
 
             for (pub_var, ret_var) in cvars.clone().iter().zip(returned_cells.unwrap()) {
                 // replace the computation of the public output vars with the actual variables being returned here
-                let var_idx = pub_var.cvar().unwrap().to_witness_var().index;
+                let var_idx = pub_var.cvar().unwrap().to_cell_var().index;
```

Comment:
I don't see why this would always be true, for example if you write something like this in noname:

```rust
fn main(pub xx: Field, yy: Field) -> Field {
   return xx + yy;
}
```

then the returned cell will be a linear combination of two vars, and the call to `to_cell_var()` will panic.

actually, let's merge this as is, and let's write a test to see if we can trigger the bug, wdyt?

Reply:
if I'm right, then the fix will probably be to reduce the linear combination to a single var before this for loop

Reply:
I thought there will be always a single var for a public output, while the `returned_cells` holds the value (xx + yy). 

I will do a check